test/setup-ca-kubernetes.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+
+# This script generates the certificates needed for securing pmem csi components.
+# This is supposed to run prior to deploying pmem-csi driver in a cluster. 
+
+generate_csr()
+{
+    CN=$1
+    CSR_NAME=${2:-$CN}
+
+    echo "Generating signing request certificate for '$CN'"
+    cert_json=$(echo '{ "CN":  "'$CN'", "key": { "algo": "ecdsa", "size": 256 } }')
+
+    echo $cert_json | cfssl genkey - | cfssljson -bare $CSR_NAME
+
+    echo "Creating kubernetes signing request: $CSR_NAME-csr"
+    kubectl delete csr $CSR_NAME-csr 2> /dev/null
+    cat <<EOF | kubectl create -f -
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: $CSR_NAME-csr
+spec:
+  groups:
+  - system:authenticated
+  request: $(cat $CSR_NAME.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+  - client auth
```

Comment:
Is all of that really needed?

Reply:
Ya not needed, limited to  'server auth' and 'client auth'.